src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -2414,6 +2414,11 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
         return false;
       }
       break;
+    case Op_VectorMaskCmp:
+      if (vlen < 2 || bit_size < 64) {
+        return false;
+      }
+      break;
```

Comment:
Hmm, currently min_vector_size for byte type is lower to 4 bytes to support vector api shuffle, but we don't have a per Opcode size check correctly. E.g. no length check for reduce_add8B, do you also see any issue for that? I think we'd better only allow 4 bytes for vector api shuffle related opcodes (and 2 for mask related opcodes) in match_rule_supported_vector, while for other opcodes, we still only support for >=64 bits.

Reply:
> Hmm, currently min_vector_size for byte type is lower to 4 bytes to support vector api shuffle, but we don't have a per Opcode size check correctly. E.g. no length check for reduce_add8B, do you also see any issue for that? 

Since SLP does not support subword reduction, I think current match rules are fine. AArch64 part looks good to me.